Admin Store Pickup Event Creation / Edit Page

WishingWell13 opened the pull request:
A bunch of file creation

<!-- Fill in all spots surrounded by brackets and confirm all check boxes after confirming completion of the tasks -->

# Info

Closes #151.

<!-- If there is no issue for this pull request yet, please create one or
delete this line if the pull request is for a very minor tweak. -->

<!-- What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context. -->

## Changes

- created the `admin/store/pickup/edit/[uuid]`, `admin/store/pickup/create` pages
- Implemented create, delete, and patch functionality for pickup events
- Added buttons to navigate to these pages on `admin/store/pickup/[uuid]` and `admin/store/pickup/`

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) <!-- Edit this type of change if you select this -->

# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally on Desktop.
- [x] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [ ] My changes produce no new warnings.
<img width="860" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/c336e5bb-7c5b-44ff-a064-ab58ea0e1d2f">

<img width="710" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/7b96ca40-7ed4-4279-8e96-3f12d4b1cdf4">

<img width="723" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/68cd35d9-1853-4e3c-9d85-ea2b75c14624">

# Screenshots

<!-- If you made any visual changes to the website, please include relevant screenshots below. -->

<img width="897" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/0f281fa2-7dae-422c-be02-01068159c509">

<img width="901" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/bf4fb5f3-ee48-4b7e-aae6-430f5ccb7182">

<img width="500" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/5dbc7fa2-b397-42ab-b449-e6814cb7fe68">

<img width="295" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/06453097-16ae-49ec-a918-a1342f6f566e">

Timer not a part of webpage

WishingWell13 committed:
UI displays, breaks when click submit

A bunch of file creation

WishingWell13 committed:
Merge branch 'main' into andy/AdminPickupEvent

alexzhang1618 committed:
Fix bugs

alexzhang1618 committed:
Merge branch 'andy/AdminPickupEvent' of https://github.com/acmucsd/membership-portal-ui-v2 into andy/AdminPickupEvent

WishingWell13 committed:
Create pickup event functionality now works

Fixed bugs around date, linkedEventUUID, and orderLimit

Note that pickup event time can still be different than the event time

Clicking "continue editing" after successful event edit not implemented yet.

WishingWell13 committed:
Deleted console.log statements

WishingWell13 committed:
Setup for Edit and Delete functionality

Fixed improperly configured apiResponse for PublicOrderPickupEvent

Setup DeletePickupEventRequest, function in the AdminEventManager, and the API call function

WishingWell13 committed:
Can now edit events

*error message seems to be broken + can't navigate to continue editing

WishingWell13 committed:
Edit page should now reroute correctly

Got rid of false popup saying that API call was correct

Order limit now autofills correctly

WishingWell13 committed:
Fixed view event page rerouting

WishingWell13 committed:
Added navigation buttons to edit and create pages

Also fixed create event page false reporting successful creations

WishingWell13 committed:
Merge branch 'main' into andy/AdminPickupEvent

WishingWell13 committed:
Merge branch 'main' into andy/AdminPickupEvent

SheepTester commented on src/pages/admin/store/pickup/index.tsx:
```diff
@@ -23,6 +24,14 @@ const AdminPickupPage = ({ futurePickupEvents, pastPickupEvents }: AdminPickupPa
+          onClick={() => router.push(`${config.admin.store.pickupCreate}`)}
+        >
+          <Typography variant="h5/bold">Create New Pickup Event</Typography>
+        </button>
```
Currently the create button is in the middle of the header, which is a bit weird. I think it makes sense to align it on the right with the `Future | Past` selector

You could do that by removing `justify-content: space-between;` from `.header`, then adding some auto margin between the two elements.

This would work:

```suggestion
        <Typography variant="h1/bold" style={{ marginRight: 'auto' }}>Manage Pickup Events</Typography>

        <button
          type="button"
          className={`${styles.displayButton} ${display === 'future' && styles.active}`}
          onClick={() => router.push(`${config.admin.store.pickupCreate}`)}
        >
          <Typography variant="h5/bold">Create New Pickup Event</Typography>
        </button>
```

SheepTester commented on src/components/admin/event/AdminPickupEvent/AdminPickupEventForm.tsx:
```diff
@@ -0,0 +1,275 @@
+            defaultValue={defaultFormText}
+            {...register('linkedEventUuid', {
+              required: 'Required',
+            })}
```
![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/0595ee7c-f404-489b-99cd-e99379beac04)

I think pickup events don't need to be linked to events, like for picking up merch on library walk

it might be nice to make this not required (by adding an option like "None"). or does the API not allow that?

SheepTester commented on src/components/admin/event/AdminPickupEvent/AdminPickupEventForm.tsx:
```diff
@@ -0,0 +1,275 @@
+      <div className={style.submitButtons}>
+        {mode === 'edit' ? (
+          <>
+            <Button submit disabled={loading}>
+              Save changes
+            </Button>
+            <Button onClick={resetForm} disabled={loading} destructive>
+              Discard changes
+            </Button>
+            <Button onClick={deletePickupEvent} disabled={loading} destructive>
+              Delete pickup event
+            </Button>
```
![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/c2e6c74e-6a7f-42a2-8430-c1e259e0e831)

It seems like it's possible to cancel and complete pickup events. I know it's a bit outside the scope for creating/editing pickup events, but maybe you could add buttons for those here and/or on the pickup event page. hopefully it's not too much extra work

you can leave this up to a future PR if you'd like, though, so we can get this merged in

WishingWell13 replied:
Let's make that a new PR

SheepTester commented on src/components/admin/event/EventDetailsForm/index.tsx:
```diff
@@ -145,7 +145,8 @@ const EventDetailsForm = (props: IProps) => {
             text: 'View Event',
-            onClick: () => router.push(`https://acmucsd.com/events/${event.uuid}`),
+            onClick: () =>
+              router.push(`https://acmucsd.com/${config.admin.store.pickup}/${event.uuid}`),
```
I think you changed the wrong file; this is for normal events

SheepTester commented on src/components/admin/event/LinkedEventSelector/LinkedEventSelector.tsx:
where is this component used

WishingWell13 replied:
Removed

SheepTester commented on src/lib/config.ts:
```diff
@@ -51,6 +51,7 @@ const config = {
           future: '/merch/order/pickup/future',
           past: '/merch/order/pickup/past',
           single: '/merch/order/pickup',
+          cleanup: '/merch/order/cleanup',
```
oh what is this for

WishingWell13 replied:
https://documenter.getpostman.com/view/12949536/TVRedVwQ#12b6ff8d-f21c-4f27-a1f3-65eb0d2a90c0

Saw this endpoint and threw it in

SheepTester commented on src/lib/managers/AdminEventManager.ts:
```diff
@@ -106,6 +112,58 @@ export const editEvent = async (data: EditEventRequest & AuthAPIHandlerProps<Pub
+  if (data.cover && data.cover.size > config.file.MAX_EVENT_COVER_SIZE_KB * 1024) {
+    onFailCallback?.(new Error('Cover size too large'));
+    return;
+  }
```
Pickup events don't have covers, I think, so you could remove this

SheepTester commented on src/lib/managers/AdminEventManager.ts:
```diff
@@ -106,6 +112,58 @@ export const editEvent = async (data: EditEventRequest & AuthAPIHandlerProps<Pub
+      // There's some weird behavior that happens when we editEvent after uploading a new
+      // event image, so I've kept the API calls in the same order as createNewEvent
+      await EventAPI.uploadEventImage(token, uuid, data.cover);
+    }
```
this too

SheepTester commented on src/pages/admin/store/pickup/create.tsx:
```diff
@@ -0,0 +1,44 @@
+
+const getServerSidePropsFunc: GetServerSideProps = async ({ req, res }) => {
+  const token = CookieService.getServerCookie(CookieType.ACCESS_TOKEN, { req, res });
+  const [futureEvents] = await Promise.all([EventAPI.getAllFutureEvents()]);
```
you dont need `Promise.all` here since you only have one promise

```suggestion
  const futureEvents = await EventAPI.getAllFutureEvents();
```

SheepTester commented on src/components/admin/event/AdminPickupEvent/style.module.scss:
```diff
@@ -0,0 +1,94 @@
+      }
+    }
+  }
+}
```
this isnt used on your page so I think you can remove it

SheepTester commented on src/components/admin/event/AdminPickupEvent/AdminPickupEventForm.tsx:
```diff
@@ -0,0 +1,275 @@
+        },
+      });
+    } catch (error) {
+      reportError('Could not delete collection', error);
```
```suggestion
      reportError('Could not delete pickup event', error);
```

SheepTester reviewed (approved):
looks good, just some nitpicks

WishingWell13 committed:
Update src/pages/admin/store/pickup/index.tsx

Currently the create button is in the middle of the header, which is a bit weird. I think it makes sense to align it on the right with the Future | Past selector

You could do that by removing justify-content: space-between; from .header, then adding some auto margin between the two elements.

Co-authored-by: Sean <ussa.say@gmail.com>

WishingWell13 committed:
Update src/components/admin/event/AdminPickupEvent/AdminPickupEventForm.tsx

Co-authored-by: Sean <ussa.say@gmail.com>

WishingWell13 committed:
Update src/pages/admin/store/pickup/create.tsx

you dont need Promise.all here since you only have one promise

Co-authored-by: Sean <ussa.say@gmail.com>

WishingWell13 committed:
Merge branch 'main' into andy/AdminPickupEvent

WishingWell13 committed:
Pickup events don't have covers, so removed checks

WishingWell13 committed:
Merge branch 'andy/AdminPickupEvent' of https://github.com/acmucsd/membership-portal-ui-v2 into andy/AdminPickupEvent

WishingWell13 committed:
Removed unused files, reverted events form

WishingWell13 committed:
Removed mandatory requirement for linked event

Also fixed linting errors and removed unnecessary css
